### PR TITLE
Matrix copy

### PR DESCRIPTION
**Pull request by sebpuetz**
Not quite #17 but still something that might be nice to have. Mirrors the implementation in `finalfusion-python`, where I stole the `copy_storage_to_array` fn from.

### CODE REVIEW

**Comment by sebpuetz** [comment]
We might also want to think about splitting `finalfusion.h` in a future PR. As of now, there's a lot going on in there.

File: tests/matrix_copy.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    if (*(embedding + i) != *(embedding_from_copy + i)) {
+      return 1;
+    }
+  }
+  free(embedding_from_copy);
+  free(embedding);
+  free(matrix_copy);
+
+  /* oov */
+  embedding = ff_embedding_lookup(embeds, "Tübingen");
```
Comment (danieldk) [inline tests/matrix_copy.c]: Does not seem to be related to matrix copying...

**Comment by sebpuetz** [comment]
I'll merge once CI is happy
